pallets/apollo-platform/src/migrations.rs

```diff
@@ -1,44 +1,89 @@
-                            .unwrap_or_else(Zero::zero);
-
-                    // Add the additional collateral to the current total
-                    let updated_total_collateral =
-                        current_total_collateral.saturating_add(additional_collateral);
-
-                    // Update or insert the new total collateral amount
-                    <UserTotalCollateral<T>>::insert(
-                        user.clone(),
-                        collateral_asset,
-                        updated_total_collateral,
-                    );
-                }
-            });
-    });
-
-    let total_migrated_entries = <UserTotalCollateral<T>>::iter().count();
-    log::info!(
-        "Migrated {} user total collateral entries",
-        total_migrated_entries
-    );
-
-    Ok(())
+pub struct MigrateToV1<T>(core::marker::PhantomData<T>);
+
+impl<T> OnRuntimeUpgrade for MigrateToV1<T>
+where
+    T: Config,
+{
+    fn on_runtime_upgrade() -> frame_support::weights::Weight {
+        if Pallet::<T>::on_chain_storage_version() != StorageVersion::new(0) {
+            error!(
+                "Runtime upgrade executed with wrong storage version, expected 0, got {:?}",
+                Pallet::<T>::on_chain_storage_version()
+            );
+            return <T as frame_system::Config>::DbWeight::get().reads(1);
+        }
+
+        info!("Applying migration to version 2: Convert borrowing info to total collateral");
+
+        // Perform migration
+        <UserBorrowingInfo<T>>::iter().for_each(|(_, user, old_borrowing_map)| {
+            old_borrowing_map
+                .iter()
+                .for_each(|(collateral_asset, borrow_info)| {
+                    let additional_collateral = borrow_info.collateral_amount;
+
+                    if additional_collateral > Balance::zero() {
+                        let current_total_collateral =
+                            <UserTotalCollateral<T>>::get(&user, collateral_asset)
+                                .unwrap_or_else(Zero::zero);
+
+                        let updated_total_collateral =
+                            current_total_collateral.saturating_add(additional_collateral);
+
+                        <UserTotalCollateral<T>>::insert(
+                            user.clone(),
+                            collateral_asset,
+                            updated_total_collateral,
+                        );
+                    }
+                });
+        });
+
+        let total_migrated_entries = <UserTotalCollateral<T>>::iter().count();
+        info!(
+            "Migrated {} user total collateral entries",
+            total_migrated_entries
+        );
+
+        // Update storage version
+        StorageVersion::new(1).put::<Pallet<T>>();
+
+        // Calculate and return weight
+        <T as frame_system::Config>::DbWeight::get().reads_writes(
+            total_migrated_entries as u64 * 2, // read old and new storage
```

Comment:
Total entries calculation also reads all storage items
```suggestion
            total_migrated_entries as u64 * 3, // read old and new storage
```